### PR TITLE
jsonnet: unlock dependencies for 4.9 development cycle

### PR DESCRIPTION
**Pull request by paulfantom**
Unlocking all jsonnet deps. All relevant work was done in `jsonnet/` directory. `assets/` and `manifests/` are generated.
Additional work to allow using latest libs:
- thanos ruler and querier need `targetGroups` during mixin inclusion
- kube-prometheus components need an explicit parameter for kube-rbac-proxy image
- removed `securityContext` from thanos querier pods. It wasn't included earlier and with one provided by the upstream community, querier cannot start. This is due to OpenShift constraints.

Some jsonnet repositories changed place or default branch:
- etcd migrated a long time ago to `etcd-io/etcd`
- etcd default branch is now `main`
- thanos default branch is now `main`
- kube-thanos default branch is now `main`

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

@prashbnair please verify if this is also bringing in relevant changes from kubernetes-mixin

/cc @simonpasquier

### CODE REVIEW

**Comment by prashbnair** [comment]
@paulfantom My changes are in.

**Comment by paulfantom** [comment]
CI was filing due to `securityContext` set on thanos querier pods. The event for that was:
```
2m43s       Warning   FailedCreate             replicaset/thanos-querier-877fd4494                Error creating: pods "thanos-querier-877fd4494-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted: .spec.securityContext.fsGroup: Invalid value: []int64{65534}: 65534 is not an allowed group, spec.containers[0].securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000440000, 1000449999], spec.containers[1].securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000440000, 1000449999], spec.containers[2].securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000440000, 1000449999], spec.containers[3].securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000440000, 1000449999], spec.containers[4].securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000440000, 1000449999], provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]  
```

File: jsonnet/jsonnetfile.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
           "subdir": "jsonnet/prometheus-operator"
         }
       },
-      "version": "release-0.47"
+      "version": "master"
```
[the remark this answers is not in the document]
Comment (paulfantom) [reply jsonnet/jsonnetfile.json:29]: CRDs and prometheus-operator alerts are coming from the same place and are locked with this `version` flag. Having this set to `master` allows us to test new alerts and ensure CRDs are really backward compatible as they should.

I agree that this should be set to particular released version of prometheus-operator when we lock all dependencies before releasing OpenShift.

File: assets/thanos-querier/deployment.yaml [hunk 2]
```diff
@@ -181,6 +187,8 @@ spec:
           name: secret-thanos-querier-tls
         - mountPath: /etc/kube-rbac-proxy
           name: secret-thanos-querier-kube-rbac-proxy-rules
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (dgrisonnet) [inline assets/thanos-querier/deployment.yaml:191]: we most likely don't want that

File: assets/thanos-querier/deployment.yaml [hunk 3]
```diff
@@ -46,13 +46,19 @@ spec:
         - --query.replica-label=prometheus_replica
         - --query.replica-label=thanos_ruler_replica
         - --store=dnssrv+_grpc._tcp.prometheus-operated.openshift-monitoring.svc.cluster.local
+        - --query.auto-downsampling
```
Comment (dgrisonnet) [inline assets/thanos-querier/deployment.yaml:49]: any concerns regarding enabling auto-downsampling?

Comment (paulfantom) [reply assets/thanos-querier/deployment.yaml:49]: I don't have any, it should reduce the load in some situations.

File: assets/thanos-querier/deployment.yaml (hunk 2 above)
Comment (paulfantom) [reply assets/thanos-querier/deployment.yaml:191]: We are not platform agnostic. Thanos is built only for linux, same for [prometheus](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/prometheus-k8s/prometheus.yaml#L169-L170), [alertmanager](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/alertmanager/alertmanager.yaml#L106-L107), [ksm](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/kube-state-metrics/deployment.yaml#L105-L106), and all other components:
```console
$ grep nodeSelector -A1 -R . 
./alertmanager/alertmanager.yaml:  nodeSelector:
./alertmanager/alertmanager.yaml-    kubernetes.io/os: linux
--
./grafana/deployment.yaml:      nodeSelector:
./grafana/deployment.yaml-        beta.kubernetes.io/os: linux
--
./kube-state-metrics/deployment.yaml:      nodeSelector:
./kube-state-metrics/deployment.yaml-        kubernetes.io/os: linux
--
./node-exporter/daemonset.yaml:      nodeSelector:
./node-exporter/daemonset.yaml-        kubernetes.io/os: linux
--
./openshift-state-metrics/deployment.yaml:      nodeSelector:
./openshift-state-metrics/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-adapter/deployment.yaml:      nodeSelector:
./prometheus-adapter/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-k8s/prometheus.yaml:  nodeSelector:
./prometheus-k8s/prometheus.yaml-    kubernetes.io/os: linux
--
./prometheus-operator-user-workload/deployment.yaml:      nodeSelector:
./prometheus-operator-user-workload/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-operator/deployment.yaml:      nodeSelector:
./prometheus-operator/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-user-workload/prometheus.yaml:  nodeSelector:
./prometheus-user-workload/prometheus.yaml-    kubernetes.io/os: linux
--
./telemeter-client/deployment.yaml:      nodeSelector:
./telemeter-client/deployment.yaml-        beta.kubernetes.io/os: linux
--
./thanos-querier/deployment.yaml:      nodeSelector:
./thanos-querier/deployment.yaml-        beta.kubernetes.io/os: linux
```

Comment (dgrisonnet) [reply assets/thanos-querier/deployment.yaml:191]: Oh ok, I thought they could be scheduled on any nodes and the respective runtimes would handle the rest.

File: jsonnet/jsonnetfile.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonpasquier) [reply jsonnet/jsonnetfile.json:29]: ok thanks for the explanation. I'm fine with this approach :)

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**